sfa_dash/blueprints/admin.py

```diff
@@ -233,11 +241,14 @@ def get(self, uuid, **kwargs):
             user_list = users.list_metadata().json()
             user_map = {user['user_id']: user
                         for user in user_list}
-            role['users'] = {k: {'user_id': k,
-                                 'added_to_user': v,
-                                 'organization': user_map.get(
-                                     k, {}).get('organization', '')}
-                             for k, v in role['users'].items()}
+            role['users'] = {
+                uid: {'user_id': uid,
+                      'added_to_user': user,
+                      'email': user_map.get(uid, {}).get(
+                          'email', self.get_email(uid)),
+                      'organization': user_map.get(uid, {}).get(
+                          'organization', 'Organization Unavailable')}
+                for uid, user in role['users'].items()}
```

Comment:
As I was recently reminded, even though we might not use the result of ``self.get_email(uid)``  the function is still executed and makes an unnecessary call to the API for every uid. We probably want to write out the for loop to get the email from user_map first, and if it isn't found call ``self.get_email``.

```python
for uid, user in role['users'].items():
    email = user_map.get(uid, {}).get('email', None)
    if email is None:
        email = self.get_email(uid)
    ...
```